<a href="https://colab.research.google.com/github/AdrNicole/DATA-ANALYTICS/blob/Modulo2_Programacion-con-Python-para-el-analisis-de-datos/D2_Clase3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Ejercicio1

a. Imprime el quinto elemento de la lista.

b. Imprime del cuarto al sexto inclusive.

c. Elimina el elemento en la posición 3

d. Agrega dos nuevos elementos a la lista

e. Ordena la lista alfabéticamente.

f. Genera un diccionario usando lista1 como llaves y lista como valores.

In [ ]:
lista = ["manzana", "pera", "naranja", "plátano", "kiwi", "uva", "piña",
"mango", "sandía","melón"]

print('-'*40,'a')
a = lista[4]
print(a)

print('-'*40,'b')
b = lista[3:5]
print(b)

print('-'*40,'c')
c = lista.remove(lista[2])
print(lista)

print('-'*40,'d')
d1 =lista.append('carambola')
d2 =lista.append('chirimoya')
print(lista)

print('-'*40,'e')
e = sorted(lista, key = lambda w: w)
print(e)

print('-'*40,'f')
lista1 = ["uno", "dos", "tres", "cuatro", "cinco", "seis", "siete",
"ocho", "nueve", "diez"]
f = dict(zip(lista1,e))
print(f)


---------------------------------------- a
kiwi
---------------------------------------- b
['plátano', 'kiwi']
---------------------------------------- c
['manzana', 'pera', 'plátano', 'kiwi', 'uva', 'piña', 'mango', 'sandía', 'melón']
---------------------------------------- d
['manzana', 'pera', 'plátano', 'kiwi', 'uva', 'piña', 'mango', 'sandía', 'melón', 'carambola', 'chirimoya']
---------------------------------------- e
['carambola', 'chirimoya', 'kiwi', 'mango', 'manzana', 'melón', 'pera', 'piña', 'plátano', 'sandía', 'uva']
---------------------------------------- f
{'uno': 'carambola', 'dos': 'chirimoya', 'tres': 'kiwi', 'cuatro': 'mango', 'cinco': 'manzana', 'seis': 'melón', 'siete': 'pera', 'ocho': 'piña', 'nueve': 'plátano', 'diez': 'sandía'}


#Ejercicio 2
Fabricación de cerveza artesanal

El objetivo del ejercicio es calcular la media, la desviación estándar, el LIC y el LSC del
porcentaje de alcohol y guardarlos en variables.
Para asegurarte de que los lotes de cerveza se estén elaborando de forma consistente y que el
nivel de alcohol sea el adecuado, una vez calculadas las medidas anteriores, indicar cuáles
valores quedarían fuera de los límites de control y cuántos son.


In [ ]:
import numpy as np
import pandas as pd
#nombres=np.loadtxt("alcohol_cerveza.csv",delimiter=',',dtype=str,encoding='utf-8')
#ruta = '/alcohol_cerveza.csv'
#df = pd.read_csv(ruta)

array = np.loadtxt('/alcohol_cerveza.csv')
print(array)

media = np.mean(array)
desviacion =np.std(array)
lic = media - (3*desviacion)
lsc = media + (3*desviacion)
condicion1 = array< lic
condicion2 = array>lsc
valores_fuera = array[np.logical_or(condicion1,condicion2)]
#fuera_de_control = quality_column[np.logical_or(quality_column < lic, quality_column > lsc)]

print(f'la media es: {media}')
print(f'desviacion es: {desviacion}')
print(f'el límite inferior de control (LIC): {lic}')
print(f'límite superior de control (LSC): {lsc}')
print('Valores fuera de control:')
print(valores_fuera if valores_fuera.size > 0 else "No se encontraron valores fuera de control.")

[0.   5.06 4.29 ... 4.68 5.12 4.92]
la media es: 5.005334665334665
desviacion es: 0.34934892044901755
el límite inferior de control (LIC): 3.9572879039876123
límite superior de control (LSC): 6.053381426681717
Valores fuera de control:
[0.   3.87]


De lo anterior se muestra que los datos deberian estar en el rango que va de 3.957 a 6.053 unidades porcentuales de alcohol de cada lote. Dados los resultados hay dos valores atípicos extremos. Ambos se encuentran por debajo del limite inferior de control x1=0 y x2=3.87 lo que afecta el promedio que fue de 5.005 unidades porcentuales. Es así, que se puede considerar
que el proceso está fuera de control y se requiere una acción correctiva para investigar y corregir la causa de la variación.